src/assembly/datasource.ts

```diff
@@ -0,0 +1,45 @@
+import * as host from "./hypermode";
+import { JSON } from "json-as";
+
+
+@json
+class pgInput {
+  query!: string;
+  params!: string[];
+}
+
+
+@json
+class pgResponse {
+  result!: string;
+  error!: string;
+}
+
+export abstract class postgresql {
+  static query(
+    datasourceName: string,
+    query: string,
+    params: string[],
```

Comment:
Params can be optional.
```suggestion
    params: string[] = [],
```

Reply:
... or preferably if you implement `QueryParameters` as mentioned in my feedback on the example, then it would be:

```suggestion
    params: QueryParameters | null = null,
```